src/libs/geotagging.c

```diff
@@ -351,6 +353,72 @@ static void _refresh_images_displayed_on_track(const int segid, const gboolean a
     }
   }
 }
+static void _update_nb_images(dt_lib_module_t *self)
+{
+  dt_lib_geotagging_t *d = (dt_lib_geotagging_t *)self->data;
+  GtkTreeModel *model = gtk_tree_view_get_model(GTK_TREE_VIEW(d->map.gpx_view));
+  GtkTreeIter iter;
+  gboolean valid = gtk_tree_model_get_iter_first(model, &iter);
+  int nb_imgs = 0;
+  for(int segid = 0; valid && segid < d->map.nb_tracks; segid++)
+  {
+    gboolean active;
+    int nb;
+    gtk_tree_model_get(model, &iter, DT_GEO_TRACKS_ACTIVE, &active,
+                                     DT_GEO_TRACKS_IMAGES, &nb, -1);
+    if(active)
+      nb_imgs += nb;
+    valid = gtk_tree_model_iter_next(model, &iter);
+  }
+  d->map.nb_imgs = nb_imgs;
+  gchar *nb = dt_util_dstrcat(NULL, "%d/%d", nb_imgs, d->nb_imgs);
+  gtk_label_set_text(GTK_LABEL(d->map.nb_imgs_label), nb);
+  g_free(nb);
+}
+
+static void _update_buttons(dt_lib_module_t *self)
+{
+  dt_lib_geotagging_t *d = (dt_lib_geotagging_t *)self->data;
+  gtk_widget_set_sensitive(d->map.preview_button, d->map.nb_tracks);
+  GtkWidget *label = gtk_bin_get_child(GTK_BIN(d->map.apply_gpx_button));
+  gtk_label_set_text(GTK_LABEL(label), d->offset ? _("apply offset and geo-location")
+                                                 : _("apply geo-location"));
+  gtk_widget_set_tooltip_text(d->map.apply_gpx_button,
+                              d->offset ? _("apply offset and geo-location to matching images"
+                                            "\ndouble operation: two ctrl-Z to undo")
```

Comment:
I'm pretty sure we can do better. Wrapping the changes into a group should do the trick and require a single undo. Can you try this ?

Reply:
I thought about this but found it a bit tricky.
First a bit of code duplication. In the job for set geolocation we have to add set offset treatment and tweak the messages accordingly.
But then the messages of undo will not accurate anymore (as specific for each operation).
Last a small doubt maybe not founded. If the order for the undos execution is not important, the order for redos is fundamental (set offset before before set geolocation). Should work ?

Still motivated ?